lmdeploy/pytorch/check_env/__init__.py

```diff
@@ -87,6 +87,27 @@ def check_env(device_type: str):
 MAX_TRANSFORMERS_VERSION = '4.41.2'
 
 
+def check_awq(hf_config):
+    """check awq support."""
+    logger = get_logger('lmdeploy')
+    quantization_config = getattr(hf_config, 'quantization_config', dict())
+    quant_method = quantization_config.get('quant_method', None)
+    if quant_method != 'awq':
+        return
+    try:
+        import awq  # noqa
+    except Exception as e:
+        _handle_exception(e, 'awq', logger)
+
+    try:
+        import awq_ext  # noqa
+    except Exception:
+        logger.debug('Exception:', exc_info=1)
```

Comment:
for cuda12 `pip install autoawq`
for cuda118 `pip install autoawq@https://github.com/casper-hansen/AutoAWQ/releases/download/v0.2.5/autoawq-0.2.5+cu118-cp38-cp38-linux_x86_64.whl`

Reply:
cuda is not the only problem.
I would rather force user to build it by themself (until we can provide our kernel).

Reply:
Let make the hint clearer, for instance, "Please ensure "autoawq" has been installed correctly", since users might "pip install awq" if they encounter the following error
```
2024-07-23 09:33:41,849 - lmdeploy - ERROR - ModuleNotFoundError: No module named 'awq'
2024-07-23 09:33:41,849 - lmdeploy - ERROR - <awq> test failed!
Please ensure it has been installed correctly.
```